app/models/option_sets/import.rb

```diff
@@ -29,24 +35,73 @@ def cleaner
     end
 
     def create_option_set_and_copy_any_errors(headers, geographic)
-      self.option_set = OptionSet.create(
+      params = {
         mission: mission,
         name: name,
-        levels: headers.map { |h| OptionLevel.new(name: h) },
         geographic: geographic,
         allow_coordinates: geographic,
         root_node: OptionNode.new
-      )
+      }
+
+      params[:levels] = headers.map { |h| OptionLevel.new(name: h) } if headers.any?
+
+      self.option_set = OptionSet.create(params)
       copy_validation_errors_for_row(1, option_set.errors) unless option_set.valid?
     end
 
-    def process_row(row, row_idx)
+    def process_row(row, row_idx, languages)
       # Metadata about the option is included at the end of the row array.
       metadata = row.extract_options!
+      if languages.any?
+        translation_row(row, row_idx, metadata, languages)
+      else
+        multilevel_row(row, row_idx, metadata)
+      end
+    end
 
+    def translation_row(row, row_idx, metadata, languages)
+      option = create_translated_option(row, metadata, languages)
+      if option.invalid?
+        copy_validation_errors_for_row(metadata[:orig_row_num], option.errors)
+        nil
+      else
+        create_translated_option_node(option, row_idx)
+        option
+      end
+    end
+
+    def create_translated_option(row, metadata, languages)
+      option = {
+        mission: mission,
+        canonical_name: row[0],
+        name_translations: build_translation_json(row, languages)
+      }
+      option[:value] = metadata[:value] if metadata[:value].present?
```

Comment:
Why create this new special header? It seems fine to do, just wondering why it was needed.

Reply:
I think....because the way the data cleaner was working, it wasn't included in the row data. I could be wrong, do you want me to look into it next week?

Reply:
No need to spend more time on it, as long as there's no underlying reason like "client asked for this specifically" then I'm happy with the paper trail we've got here. If this somehow becomes an issue later, we can update it as needed.